Psychtoolbox/PsychDemos/DrawFormattedText2Demo.m

```diff
@@ -0,0 +1,138 @@
+% DrawFormattedText2Demo
+%
+% Draws formatted text, shows how to align and center text vertically
+% and/or horizontally, how line-breaks are introduced, how to transform
+% text.
+%
+% Press any key to cycle through different demo displays.
+%
+% see also: PsychDemos, Screen DrawText?, DrawSomeTextDemo, DrawFormattedText2Demo
+
+% 09/15/16    dcn    Wrote it.
+
+try
+    AssertOpenGL;   % check PTB is functioning
+    
+    % Choosing the display with the highest display number is
+    % a best guess about where you want the stimulus displayed.
+    screens=Screen('Screens');
+    screenNumber=max(screens);
+
+    % Open window with default settings:
+    PsychImaging('PrepareConfiguration');
+    % make screen partially transparent on OSX and windows vista or
+    % higher, so we can debug.
+%    PsychDebugWindowConfiguration;
+    [w,wrect]=PsychImaging('OpenWindow', screenNumber, 128);
+    
+    % This preference setting selects the high quality text renderer 
+    % each operating system: It is not really needed, as the high quality
+    % renderer is the default on all operating systems, so this is more of
+    % a "better safe than sorry" setting. DrawFormattedText2 _only_ works
+    % with this renderer. So it fails to load (see the warnings on the
+    % command line, the output of this demo will look horrible)
+    Screen('Preference', 'TextRenderer', 1);
```

Comment:
I would add a check to DrawFormattedText2 a la Screen('Preference','TextRenderer') == 1
If a different text renderer is selected you can error out. If the plugin fails to load, 'TextRenderer' will also return 0 instead of 1 afaik, so you get proper error handling instead of surprises.